manager/integration/tests/test_scheduling.py

```diff
@@ -908,13 +904,15 @@ def test_data_locality_basic(client, core_api, volume_name, pod, settings_reset)
 
     for replica in volume4.replicas:
         wait_for_replica_running(client, volume4_name, replica["name"])
+
+    volume4 = client.by_id_volume(volume4_name)
+    for replica in volume4.replicas:
```

Comment:
Get the updated list of replica before checking its host. Otherwise, the `hostId` might be empty